js/net.ts

```diff
@@ -12,7 +12,10 @@ export type Network = "tcp";
 // export type Network = "tcp" | "tcp4" | "tcp6" | "unix" | "unixpacket";
 
 // TODO Support finding network from Addr, see https://golang.org/pkg/net/#Addr
-export type Addr = string;
+export interface Addr {
+  network: Network;
+  address: string;
+}
```

Comment:
Can you please leave the Addr changes out of this PR? I'd like to discuss them separately from shutdown.